scipy/stats/_qmc.py

```diff
@@ -0,0 +1,1587 @@
+    """Optimal design.
+
+    Optimize the design by doing random permutations to lower the centered
+    discrepancy. If `optimization` is False, `niter` design are generated and
+    the one with lowest centered discrepancy is return. This option is faster.
+    Otherwise,
+
+    Centered discrepancy-based design shows better space filling robustness
+    toward 2D and 3D subprojections. Distance-based design shows better space
+    filling but less robustness to subprojections.
+
+    Parameters
+    ----------
+    d : int
+        Dimension of the parameter space.
+    start_design : array_like (n, d), optional
+        Initial design of experiment to optimize.
+    niter : int
+        Number of iterations to perform. Default is 1.
+    force : bool
+        If `optimization`, force *basinhopping* optimization. Otherwise
+        grid search is used. Default is False.
+    optimization : bool
+        Optimal design using global optimization or random generation of
+        `niter` samples. Default is True.
+    seed : {int or `numpy.random.RandomState` instance}, optional
+        If `seed` is not specified the `numpy.random.RandomState`
+        singleton is used.
+        If `seed` is an int, a new ``RandomState`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``RandomState`` instance, then that
+        instance is used.
+
+    References
+    ----------
+    .. [1] Fang et al. Design and modeling for computer experiments,
+       Computer Science and Data Analysis Series, 2006.
+    .. [2] Damblin et al., "Numerical studies of space filling designs:
+       optimization of Latin Hypercube Samples and subprojection properties",
+       Journal of Simulation, 2013.
+
+    Examples
+    --------
+    Generate samples from an optimal design.
+
+    >>> from scipy.stats import qmc
+    >>> sampler = qmc.OptimalDesign(d=2, seed=12345)
+    >>> sample = sampler.random(n=5)
+    >>> sample
+    array([[0.64091206, 0.77846875],
+           [0.43678376, 0.18474763],
+           [0.18592322, 0.80535794],
+           [0.91354501, 0.27478424],
+           [0.26327511, 0.43099467]])
+
+    Compute the quality of the sample using the discrepancy criterion.
+
+    >>> qmc.discrepancy(sample)
+    0.019688311022535432
+
+    You can possibly improve the quality of the sample by performing more
+    optimization iterations by using `niter`:
+
+    >>> sampler_2 = qmc.OptimalDesign(d=2, niter=2, seed=12345)
+    >>> sample_2 = sampler_2.random(n=5)
+    >>> qmc.discrepancy(sample_2)
+    0.019607673478802434
+
+    Finally, samples can be scaled to bounds.
+
+    >>> bounds = [[0, 2], [10, 5]]
+    >>> qmc.scale(sample, bounds)
+    array([[6.40912056, 4.33540624],
+           [4.36783762, 2.55424289],
+           [1.85923219, 4.41607383],
+           [9.13545006, 2.82435271],
+           [2.63275111, 3.29298401]])
+
+    """
+
+    def __init__(self, d: int, start_design: bool = None, niter: int = 1,
```

Comment:
`= None` usually doesn't make sense for boolean parameters, should this be True or False? If it doesn't just default to True or False, then the annotation should be `Optional[bool]`.

Reply:
Sorry it was a typo. I changed it to `Optional[npt.ArrayLike] = None`